Makefile

```diff
@@ -1,13 +1,24 @@
-package: 
+.PHONY: clippy dev format lint package run scratch test ts_test test_application test_rust test_rehash
+
+ifeq ($(OSTYPE),darwin)
+PACKAGE := application/release/mac/WereSoCool.app
+else
+PACKAGE := application/release/linux-unpacked/weresocool
+endif
+
+$(PACKAGE): application/node_modules
 	cd application && yarn package
```

Comment:
This is an example of a target that does match a file on the filesystem. With this target in place you can give `$(PACKAGE)` as a dependency to the `run` target to tell make to automatically build the package before running if it hasn't been built yet. Because `$(PACKAGE)` has a transitive dependency on `node_modules` and `yarn.lock`, running the `run` target will also rebuild the package if either of those has changed since the package was built.